kafka-rest/src/main/java/io/confluent/kafkarest/controllers/DefaultTopicConfigManagerImpl.java

```diff
@@ -0,0 +1,153 @@
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import io.confluent.kafkarest.common.KafkaFutures;
+import io.confluent.kafkarest.entities.AbstractConfig;
+import io.confluent.kafkarest.entities.BrokerConfig;
+import io.confluent.kafkarest.entities.ConfigSource;
+import io.confluent.kafkarest.entities.ConfigSynonym;
+import io.confluent.kafkarest.entities.TopicConfig;
+import kafka.server.KafkaConfig;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.CreateTopicsOptions;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.common.errors.InvalidRequestException;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Collectors;
+
+import static io.confluent.kafkarest.controllers.Entities.checkEntityExists;
+import static io.confluent.kafkarest.controllers.Entities.findEntityByKey;
+import static java.util.Collections.singletonList;
+import static java.util.Objects.requireNonNull;
+
+final class DefaultTopicConfigManagerImpl implements DefaultTopicConfigManager {
+
+  private final Admin adminClient;
+  private final ClusterManager clusterManager;
+  private final BrokerConfigManager brokerConfigManager;
+  private final BrokerManager brokerManager;
+
+  @Inject
+  DefaultTopicConfigManagerImpl(Admin adminClient,
+                                ClusterManager clusterManager,
+                                BrokerConfigManager brokerConfigManager,
+                                BrokerManager brokerManager) {
+    this.adminClient = requireNonNull(adminClient);
+    this.clusterManager = requireNonNull(clusterManager);
+    this.brokerConfigManager = requireNonNull(brokerConfigManager);
+    this.brokerManager = requireNonNull(brokerManager);
+  }
+
+  @Override
+  public CompletableFuture<List<TopicConfig>> listDefaultTopicConfigs(
+      String clusterId, String topicName) {
+    return clusterManager.getCluster(clusterId)
+        .thenApply(cluster -> checkEntityExists(cluster, "Cluster %s cannot be found.", clusterId))
+        .thenCompose(
+            cluster ->
+                KafkaFutures.toCompletableFuture(
+                    adminClient.createTopics(
+                        singletonList(new NewTopic(topicName, Optional.empty(), Optional.empty())),
+                        new CreateTopicsOptions().validateOnly(true)
+                    ).config(topicName)))
+        .handle((result, ex) -> {
+              // InvalidRequestException can happen in the case of multi-tenancy, where passing
```

Comment:
`multi-tenancy` is not available in AK. Maybe we should remove its reference from the comment.